### PR TITLE
Implement remove-thumbnails command

### PR DESCRIPTION
**Pull request by MAXakaWIZARD**
With this command we can remove thumbnails of specified format or even all thumbnails.
This one will remove all thumbs of `small` format from `default` context:
```bash
app/console sonata:media:remove-thumbnails sonata.media.provider.image default small
```
This one will remove all thumbs from `default` context:
```bash
app/console sonata:media:remove-thumbnails sonata.media.provider.image default all
```
Also, command gives ability to choose provider, context and format, if any of them are not specifed in command line.

### CODE REVIEW

**Comment by rande** [comment]
I think we need to add a remove method in the Thumbnail interface. This
will avoid duplicate code and make sure custom implementation will still
work.

**Comment by MAXakaWIZARD** [comment]
@rande done

**Comment by MAXakaWIZARD** [comment]
@rande Any news when this PR can be merged?

File: Thumbnail/FormatThumbnail.php [hunk 1]
```diff
@@ -84,18 +85,59 @@ public function generate(MediaProviderInterface $provider, MediaInterface $media
...
-                $provider->getFilesystem()->delete($path);
+        $filesystem = $provider->getFilesystem();
+
+        if (is_null($format)) {
+            //delete all thumbnails
+            $format = 'default_whateverformat';
+
+            $relPath = $provider->generatePrivateUrl($media, $format);
+            $relDir = dirname($relPath);
+            $fsRootDir = $this->getFilesystemRootDir($filesystem);
```
Comment (rande) [inline Thumbnail/FormatThumbnail.php]: You cannot do that. The filesystem can be aws. So you cannot rely on php fs functions

Comment (MAXakaWIZARD) [inline Thumbnail/FormatThumbnail.php]: You mean `dirname`?

Comment (rande) [inline Thumbnail/FormatThumbnail.php]: The finder, pathinfo ..

Comment (MAXakaWIZARD) [inline Thumbnail/FormatThumbnail.php]: So, what should I do? Leave old code and use new code only for `Local` adapter?

Comment (rande) [inline Thumbnail/FormatThumbnail.php]: I think you just need to keep the current logic, but add an extra check:

```php
    public function delete(MediaProviderInterface $provider, MediaInterface $media, array $formats = null)
    {
        $formats = $formats ?: array_keys($provider->getFormats());

        // delete the differents formats
        foreach ($formats as $format) {
            $path = $provider->generatePrivateUrl($media, $format);
            if ($path && $provider->getFilesystem()->has($path)) {
                $provider->getFilesystem()->delete($path);
            }
        }
    }
```

Comment (MAXakaWIZARD) [inline Thumbnail/FormatThumbnail.php]: @rande I don't understand, what you mean. The code you posted is original code with no changes.

Comment (rande) [inline Thumbnail/FormatThumbnail.php]: nop it is not the original, the provided formats are now specified, so you can remove specific formats

Comment (MAXakaWIZARD) [inline Thumbnail/FormatThumbnail.php]: @rande but in that way we cannot remove all old thumbnails, if we do not remember format names

Comment (rande) [inline Thumbnail/FormatThumbnail.php]: Of course you can, there is no check on the format validity. The media and the format is enough to generate the private path, and so remove it from the fs driver.

Comment (MAXakaWIZARD) [inline Thumbnail/FormatThumbnail.php]: I mean if there was some format long time ago, and i do not remember it's name, I cannot remove it, so code should search filesystem by mask.

Comment (rande) [inline Thumbnail/FormatThumbnail.php]: I think the situation is an edge case and we don't want to support this.

Comment (MAXakaWIZARD) [inline Thumbnail/FormatThumbnail.php]: @rande OK, I will come again later ;)

**Comment by MAXakaWIZARD** [comment]
@rande Please review

**Comment by MAXakaWIZARD** [comment]
@rande Any news when this PR can be merged?

**Comment by rande** [comment]
Can you add the related documentation about the new command ? also this PR introduce a BC break, can you add it to the CHANGELOG file ?

Thanks

**Comment by MAXakaWIZARD** [comment]
@rande I don't see any BC break here, it's just new param in `remove` method with default value with exactly the same behavior by default.

File: Thumbnail/ThumbnailInterface.php [hunk 2]
```diff
@@ -39,6 +39,7 @@ public function generate(MediaProviderInterface $provider, MediaInterface $media
     /**
      * @param \Sonata\MediaBundle\Provider\MediaProviderInterface $provider
      * @param \Sonata\MediaBundle\Model\MediaInterface            $media
+     * @param string|array                                        $formats
      */
-    public function delete(MediaProviderInterface $provider, MediaInterface $media);
+    public function delete(MediaProviderInterface $provider, MediaInterface $media, $formats = null);
```
Comment (rande) [inline Thumbnail/ThumbnailInterface.php]: You add new arguments in the interface, so it is a BC breaks

**Comment by MAXakaWIZARD** [comment]
@rande Ah, I see, new param in interfaces. OK

**Comment by MAXakaWIZARD** [comment]
@rande Done. Tests fail, but as I see it's not related to this PR.

**Pull request merged**